larpandoracontent/LArMonitoring/VisualParticleMonitoringAlgorithm.h

```diff
@@ -0,0 +1,105 @@
+/**
+ *  @file   larpandoracontent/LArMonitoring/VisualParticleMonitoringAlgorithm.h
+ *
+ *  @brief  Header file for the particle visualisation algorithm.
+ *
+ *  $Log: $
+ */
+#ifndef LAR_VISUAL_PARTICLE_MONITORING_ALGORITHM_H
+#define LAR_VISUAL_PARTICLE_MONITORING_ALGORITHM_H 1
+
+#include "Pandora/Algorithm.h"
+
+#include "larpandoracontent/LArHelpers/LArMCParticleHelper.h"
+
+namespace lar_content
+{
+
+/**
+ *  @brief  VisualParticleMonitoringAlgorithm class
+ */
+class VisualParticleMonitoringAlgorithm: public pandora::Algorithm
+{
+public:
+    /**
+     *  @brief  Default constructor
+     */
+    VisualParticleMonitoringAlgorithm();
+
+    virtual ~VisualParticleMonitoringAlgorithm();
+
+private:
+    pandora::StatusCode Run();
+    pandora::StatusCode ReadSettings(const pandora::TiXmlHandle xmlHandle);
+
+#ifdef MONITORING
+    /**
+     *  @brief  Visualize all MC particles independently.
+     *
+     *  This function ensures each MC particle has its own entry in the TEve tree, though common PDG colour scheme is used.
+     *
+     *  @param  mcMap The map from MC particles to calo hits
+     **/
+    void VisualizeIndependentMC(const LArMCParticleHelper::MCContributionMap &mcMap);
```

Comment:
Can any of these member functions be const?

Reply:
Yep, all of them can, will do.